versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/DatabaseAdapterConfig.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.adapter;
+
+import org.immutables.value.Value;
+
+/**
+ * Base database-adapter configuration type.
+ *
+ * <p>See {@link org.projectnessie.versioned.persist.adapter.spi.DefaultDatabaseAdapterConfig} for
+ * the default, immutable-annotated type.
+ *
+ * <p>{@link DatabaseAdapter} implementations that need more configuration options must extend this
+ * interface, have the {@link Value.Immutable} annotation and declare the {@code with...} methods
+ * implemented by "immutables".
+ */
+public interface DatabaseAdapterConfig {
+
+  int DEFAULT_PARENTS_PER_COMMIT = 20;
```

Comment:
nit the `int`s are constants and the `String`s arent. Would be nice to do one or the other

Reply:
Had to move the "default branch name" out of `DatabaseAdapter`, so that particular option is gone here.